server/service/src/document/encounter/encounter_schema.rs

```diff
@@ -0,0 +1,9 @@
+extern crate schemafy_core;
+extern crate serde;
+extern crate serde_json;
+
+use serde::{Deserialize, Serialize};
+
+schemafy::schemafy!("src/document/schemas/encounter.json");
```

Comment:
should this also be inserted on initialisation ?

Reply:
Yes its already inserted in `service/src/sync/init_programs_data.rs`

Reply:
Hmm why would it be needed here if we can access it through database ?

Reply:
I use the schemafy crate to get a Rust type from the schema. For example, I will need it later on to access encounter times etc. Especially useful since the schema is currently not finalized. Later on I could also just access the raw json but having a type is a bit nicer.

Reply:
> For example, I will need it later on to access encounter times 

I was asking during the meeting about static logic being attached to dynamic nature of documents. I can see that's it's kind of needed for patient (name) needing to be edited from a documents, but I thought it was up to the country to decide what encounter is ? Or is there 'base' schema that all encounters have that would need to be merged with schema that is defined for the country.

I am just flagging that in my experience it's quite problematic attaching static logic that relies on data that may or may not be there due to dynamic and configurable nature of that data. We can explicitly verify json schema has the required elements when the schema is configured, but this type of enforcing is better done with database structures imo

Reply:
yes the encounters are similar to patients, we need some common base format every encounter needs to follow otherwise we can't handle it to provide basic encounter logic. And yes similar some data of the encounter will be mirrored in a DB table to be able to work with it. Lets discuss tomorrow in the office not sure if I understood you correctly.